### ISSUE TITLE
overwrite and append mode behavior

### ISSUE THREAD

**Issue by changhiskhan** [lancedb/lance#568]
Transcribing/Paraphrasing feedback from a separate venue.

When using `lance.write_dataset` it's not obvious whether the dataset already exists. The current semantics are:

```
create - if it exists then raises an error, otherwise writes new dataset folder
overwrite - if it exists then it writes a new version, otherwise raises error
append - if it exists then it writes a new version, otherwise raises error
```

A user reported that "if the file does not exist, I would expect the function to create it, but instead, I see an error that the manifest doesn't exist".

An additional suggestion was "For append, I'd say issue a warning then make a new one"

**Comment by cemoody** [comment]
Can you use `write_dataset` to concatenate two Lance files, including merging vector indexes? I've got a few hundreds Lance files with ~1M vectors each. Would be great to be able to merge them without having to recompute the index

**Comment by changhiskhan** [comment]
Current state:

**Comment by cemoody** [comment]
*very* excited for both of those. The index merging enables large-scale vector search while still building the lance files in a separate nodes / machines / processes. 

And, I think, not having to call `to_table()` means I could look up individual rows across the whole partitioned dataset.... very exciting!

**Comment by changhiskhan** [comment]
oh but if you have several hundred, we may want to consider a different API for that.

e.g., we could have each process write their lance data and the index to the same destination, and then at the very end, re-write the manifest altogether.

i'll start a separate issue to discuss this

**Comment by Renkai** [comment]
If we use write append `dataset_new` on `dataset_v0`, the value of `dataset_v1` should be `dataset_v0 union dataset_new` right? Does our current implementation have a schema check before the union?

**Comment by changhiskhan** [comment]
closed via #690
